### PR TITLE
VZ-4274 Implement ConfigMap controller to update workload MutatingWebhookConfiguration

### PR DESCRIPTION
**Pull request by davisgomes**
# Description

This controller watches a specific ConfigMap for updates to the kinds of workloads enabled by the Metrics Temaplates. It updates the MutatingwebhookConfiguration to allow the webhook to process new workload kinds.

Fixes VZ-4274

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/controllers/configmap/confgimap_controller_utils_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2020, 2022, Oracle and/or its affiliates.
```
Comment (kminder) [inline application-operator/controllers/configmap/confgimap_controller_utils_test.go:1]: Just one year for new files.

File: application-operator/controllers/configmap/configmap_controller.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2020, 2022, Oracle and/or its affiliates.
```
Comment (kminder) [inline application-operator/controllers/configmap/configmap_controller.go:1]: One year for new files

File: application-operator/controllers/configmap/configmap_controller.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package configmap
```
Comment (kminder) [inline application-operator/controllers/configmap/configmap_controller.go:4]: This package name seems overly generic.

File: application-operator/controllers/configmap/configmap_controller.go (hunk 2 above)
Comment (davisgomes) [reply application-operator/controllers/configmap/configmap_controller.go:1]: Thanks. I will update on all.

File: application-operator/controllers/configmap/configmap_controller.go (hunk 3 above)
Comment (davisgomes) [reply application-operator/controllers/configmap/configmap_controller.go:4]: It is, but we did the same for namespaces, so I followed that convention

File: application-operator/controllers/configmap/configmap_controller_test.go [hunk 4]
```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2020, 2022, Oracle and/or its affiliates.
```
Comment (kminder) [inline application-operator/controllers/configmap/configmap_controller_test.go]: year

**Pull request closed without merging**